app/uk/gov/hmrc/ngrnotify/model/bridge/Bridge.scala

```diff
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ngrnotify.model.bridge
+
+import play.api.libs.json.*
+
+
+object Bridge:
+
+  // TODO We may prefer to define the Id type as a synonym for String rather than employing Scala 3 unions
+  type Id = Int | String
+
+  given Reads[Id] = new Reads[Id] {
+    def reads(jsValue: JsValue): JsResult[Id] =
+      jsValue match
+        case JsNumber(num) => JsSuccess(num.toInt)
+        case JsString(str) => JsSuccess(str)
+        case _ => JsError("Expected an Int or String")
+  }
+
+  given Writes[Id] = new Writes[Id] {
+    override def writes(intOrString: Id): JsValue =
+      intOrString match
+        case int: Int => JsNumber(int)
+        case str: String => JsString(str)
+  }
+
+  //
+  //   NOTE
+  //   -----
+  //
+  //   Note that the Bridge API engineers wanted to model the JSON `id` property as having
+  //   either JSON null or JSON number (such as 123) or JSON string (such as "123") values.
+  //
+  //   We believe that the `id` JSON values should have been modelled as having either JSON
+  //   null or JSON string value (such as "123"), simply because that would have included all
+  //   practical scenarios of usage. Therefore, rather than mirroring the Bridge API design,
+  //   and rather than having to employ Scala 3 union types, we may prefer to model our
+  //   Scala `id` counterparts as of Scala "Option[String]" type.
+  //
+  //   Therefore, the following alternative Scala code may replace the above:
+  //
+  //      ┌──────────────────────────────────────────────────────────────────┐
+  //      │ type Id = String                                                 │
+  //      │                                                                  │
+  //      │ given Reads[Id] = new Reads[Id] {                                │
+  //      │   def reads(jsValue: JsValue): JsResult[Id] =                    │
+  //      │     jsValue match                                                │
+  //      │       case JsNumber(num) => JsSuccess(num.toString)              │
+  //      │       case JsString(str) => JsSuccess(str)                       │
+  //      │       case _             => JsError("Expected an Int or String") │
+  //      │ }                                                                │
+  //      │                                                                  │
+  //      │ given Writes[Id] = new Writes {                                  │
+  //      │   override def writes(id: Id): JsValue = JsString(id)            │
+  //      │ }                                                                │
+  //      └──────────────────────────────────────────────────────────────────┘
+  //
+  //   Note that we still need to specify the Scala type Id (as synonym for String) to make
+  //   sure that Scala the compiler will pick our custom Reader from the build scope.
```

Comment:
To my understanding, we are in agreement that type `Id` should be parsed into `JsString`, as you demonstrate in the alternative code. Retaining numeric IDs as `Int` offers no tangible advantage, given that they are primarily used in request paths for subsequent requests, where the ID is automatically converted to a `String`.

Reply:
Apologies, however, if we have already reached an agreement on this matter, I do not see any reason why it should be addressed in the next PR rather than in the current one.